lib/swaps/Swaps.ts

```diff
@@ -150,6 +165,40 @@ class Swaps extends EventEmitter {
     return;
   }
 
+  /**
+   * Saves deal to database and deletes from memory.
+   * @param deal The deal to persist.
+   */
+  private persistDeal = (deal: SwapDeal) => {
+    const { r_hash } = deal;
+    if (this.hashs.has(r_hash)) {
+      const persistDeal: SwapDealAttributes = {
+        r_hash,
+        orderId: deal.orderId,
+        localOrderId: deal.localOrderId,
+        r_preimage: deal.r_preimage,
+        myRole: deal.myRole,
+        state: deal.state,
+        stateReason: deal.stateReason,
+        peerPubKey: deal.peerPubKey,
+        proposedQuantity: deal.proposedQuantity,
+        quantity: deal.quantity,
+        pairId: deal.pairId,
+        price: deal.price,
+        takerCltvDelta: deal.takerCltvDelta,
+        makerCltvDelta: deal.makerCltvDelta,
+        createTime: deal.createTime,
+        executeTime: deal.executeTime,
+        completionTime: deal.completionTime,
+      };
+      this.repository.addSwapDeal(persistDeal)
```

Comment:
We try to use the async/await pattern wherever possible so you can just await `addSwapDeal` and then await `removeDeal`.